bcda/bcdacli/cli_test.go

```diff
@@ -836,3 +834,110 @@ func getRandomPort(t *testing.T) int {
 
 	return listener.Addr().(*net.TCPAddr).Port
 }
+
+func CreateTestZipFile(file string) error {
+	zf, err := os.Create(file)
+	if err != nil {
+		return err
+	}
+	defer zf.Close()
+
+	w := zip.NewWriter(zf)
+	var files = []string{
+		"T.BCD.A0001.ZC8Y18.D181120.T1000001",
+		"T.BCD.A0001.ZC8Y18.D181120.T1000002",
+		"T.BCD.A0001.ZC8Y18.D181120.T1000003",
+	}
+
+	for _, file := range files {
+		f, err := w.Create(file)
+		if err != nil {
+			return err
+		}
+		_, err = f.Write([]byte("foo bar"))
+		if err != nil {
+			return err
+		}
+	}
+
+	return w.Close()
+}
+
+func getFileCount(path string) int {
+	f, _ := ioutil.ReadDir(path)
+	return len(f)
+}
+
+func TestCloneCCLFZips(t *testing.T) {
```

Comment:
Nit - I think we can consolidate `TestCloneCCLFZips` and `TestCloneCCLFZip`

If we test it by invoking using testApp.run (ex: https://github.com/CMSgov/bcda-app/blob/master/bcda/bcdacli/cli_test.go#L115), we can combine the assertions that you're making in both of these functions. 1: Confirming number of zip files, 2. Confirming number of subfiles within each zip.

Reply:
I consolidated this into one giant test. I'm not 100% sold on this being the cleanest way to implement a test (though I see the value in using `s.testApp.Run(args)`). But, regardless, it tests all aspects of the implementation.